### PR TITLE
break session only on illegal origin

### PR DESCRIPTION
**Pull request by DaanHoogland**
## Description

#3713 was meant to implement this but did not buid due to plugins no longer being available, hence this port forward.

Added a new CloudRuntime Exception to distinguish between command not allowed and command issued from disallowed origin, so users that are authenticated still get a proper error message instead of having their session revoked bluntly.

this was a regression in 4.11 (compared to 4.9) so it is on the 4.11 branch and will have to be merged forward.

Fixes: #3625

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

by manual testing: 
image
after changing the permission of the role user on listAccounts a regular user logs on and goes to the account tab to get the above message instead of their session being invalidated.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/api/ApiServer.java [hunk 1]
```diff
@@ -843,11 +844,7 @@ public boolean verifyRequest(final Map<String, Object[]> requestParameters, fina
             if (userId != null) {
                 final User user = ApiDBUtils.findUserById(userId);
 
-                if (!commandAvailable(remoteAddress, commandName, user)) {
-                    return false;
-                }
-
-                return true;
+                return commandAvailable(remoteAddress, commandName, user);
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/api/ApiServer.java]: :+1:

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@borisstoyanov I'm inclined to merge. Do you think the HA errors above can be in any way related?

**Comment by borisstoyanov** [comment]
@DaanHoogland I don't see how they can be related, but let's do another round maybe? 
@blueorangutan test

**Comment by DaanHoogland** [comment]
merged. tested and reviewed.

**Pull request merged**